From: Ben Hutchings <ben@decadent.org.uk>
Subject: alx: remove atl1c devices
Date: Sun, 27 Jan 2013 23:37:52 +0000
Bug-Debian: http://bugs.debian.org/699129
Forwarded: not-needed

When alx was first reviewed, upstream demanded that any changes to
support for the currently supported AR8131/8132/8151/8152 chips be
made as a series of patches, not by deprecating or removing atl1c in a
monolithic patch.  Therefore:

Luis R. Rodriguez <mcgrof@frijolero.org> wrote:
> The alx driver is to only support the AR8161 and AR8162
> devices. The older devices are supported through atl1c.

I've also changed the Kconfig help text to reflect this.

---
--- a/drivers/net/ethernet/atheros/Kconfig
+++ b/drivers/net/ethernet/atheros/Kconfig
@@ -56,14 +56,13 @@ config ATL1E
 	  will be called atl1e.
 
 config ATL1C
-	tristate "Atheros L1C Gigabit Ethernet support (DEPRECATED)"
+	tristate "Atheros L1C Gigabit Ethernet support"
 	depends on PCI
 	select CRC32
 	select NET_CORE
 	select MII
 	---help---
 	  This driver supports the Atheros L1C gigabit ethernet adapter.
-	  This driver is deprecated in favor for the alx (CONFIG_ALX) driver.
 	  This driver supports the following chipsets:
 
 	  1969:1063 - AR8131 Gigabit Ethernet
@@ -84,18 +83,7 @@ config ALX
 	select MII
 	---help---
 	  This driver supports the Atheros L1C/L1D/L1F gigabit ethernet
-	  adapter. The alx driver is intended to replace completely the
-	  atl1c driver with proper support and commitment from Qualcomm
-	  Atheros (QCA). Both atl1c and alx supports the following chipsets:
-
-	  1969:1063 - AR8131 Gigabit Ethernet
-	  1969:1062 - AR8132 Fast Ethernet (10/100 Mbit/s)
-	  1969:2062 - AR8152 v2.0 Fast Ethernet
-	  1969:2060 - AR8152 v1.1 Fast Ethernet
-	  1969:1073 - AR8151 v1.0 Gigabit Ethernet
-	  1969:1083 - AR8151 v2.0 Gigabit Ethernet
-
-	  Only alx supports the following chipsets:
+	  adapter. alx supports the following chipsets:
 
 	  1969:1091 - AR8161
 	  1969:1090 - AR8162
--- a/drivers/net/ethernet/atheros/alx/alx_main.c
+++ b/drivers/net/ethernet/atheros/alx/alx_main.c
@@ -33,12 +33,6 @@ static const char alx_drv_description[]
 #define ALX_ETHER_DEVICE(device_id) {\
 	PCI_DEVICE(ALX_VENDOR_ID, device_id)}
 static DEFINE_PCI_DEVICE_TABLE(alx_pci_tbl) = {
-	ALX_ETHER_DEVICE(ALX_DEV_ID_AR8131),
-	ALX_ETHER_DEVICE(ALX_DEV_ID_AR8132),
-	ALX_ETHER_DEVICE(ALX_DEV_ID_AR8151_V1),
-	ALX_ETHER_DEVICE(ALX_DEV_ID_AR8151_V2),
-	ALX_ETHER_DEVICE(ALX_DEV_ID_AR8152_V1),
-	ALX_ETHER_DEVICE(ALX_DEV_ID_AR8152_V2),
 	ALX_ETHER_DEVICE(ALX_DEV_ID_AR8161),
 	ALX_ETHER_DEVICE(ALX_DEV_ID_AR8162),
 	{0,}
